src/machine/i2s.go

```diff
@@ -9,6 +9,23 @@
 
 package machine
 
+import "errors"
+
+// If you are getting a compile error on this line please check to see you've
+// correctly implemented the methods on the I2S type. They must match
+// the interface method signatures type to type perfectly.
+// If not implementing the I2S type please remove your target from the build tags
+// at the top of this file.
+var _ interface {
+	SetSampleFrequency(freq uint32) error
+	ReadMono(b []uint16) (int, error)
+	ReadStereo(b []uint32) (int, error)
+	WriteMono(b []uint16) (int, error)
+	WriteStereo(b []uint32) (int, error)
+	Paused(disabled bool)
```

Comment:
What does Paused do?

Reply:
Maybe it should be called `Enabled()`? I took that name from here https://github.com/tinygo-org/pio/blob/main/rp2-pio/piolib/i2s.go#L97 but now that you mention it I think we should change them both.

Reply:
Maybe we can omit it for now. I'm not sure what fraction of i2s hardware implements the pause functionality. Same goes for Stop... I'm not sure what Stop does exactly, especially seeing Paused _seems_ to do the same thing. I really didn't think much of the implementation through, just copy pasted the C SDK code  :sweat_smile:

Reply:
I'm also thinking Stop and Paused could be implemented by a higher-level software layer which splits up incoming buffers to allow interrupting read/writes whereas hardware read/writes would not allow pausing and stopping.

Reply:
I think we should have an `Enable()` function. But also remove `Paused()` and `Stop()` for the reasons you just expressed. What do you think?

Reply:
We can always add it later on if it is really needed, but I've got a feeling from what I can tell it may be an early abstraction to add it at the hardware level. Can we leave it out for now?

Reply:
YAGNI is usually the correct choice. OK I will remove.